src/gtest/test_keys.cpp

```diff
@@ -284,13 +321,20 @@ TEST(Keys, EncodeAndDecodeUnifiedFullViewingKeys)
             auto key = libzcash::SaplingDiversifiableFullViewingKey::Read(ss);
             ASSERT_TRUE(builder.AddSaplingKey(key));
         }
+        if (!test[2].isNull()) {
+            auto data = ParseHex(test[2].get_str());
+            ASSERT_EQ(data.size(), 96);
+            CDataStream ss(data, SER_NETWORK, PROTOCOL_VERSION);
+            auto key = libzcash::OrchardFullViewingKey::Read(ss);
+            ASSERT_TRUE(builder.AddOrchardKey(key));
+        }
 
-        // Orchard keys and unknown items are not yet supported; instead,
-        // we just test that we're able to parse the unified key string
-        // and that the constituent items match the elements; if no Sapling
-        // key is present then UFVK construction would fail because it might
-        // presume the UFVK to be transparent-only.
-        if (test[1].isNull())
+        // Unknown items are not yet supported; instead, we just test that we're
+        // able to parse the unified key string and that the constituent items
+        // match the elements; if no Sapling or Orchard key is present then UFVK
+        // construction would fail because it might presume the UFVK to be
+        // transparent-only.
```

Comment:
All of the test vectors by construction include either a Sapling or an Orchard receiver:
https://github.com/zcash-hackworks/zcash-test-vectors/blob/fe7ed09c6cde02c92457d7647a30c9127a29f3ee/zcash_test_vectors/unified_full_viewing_keys.py#L63

Reply:
I think the comment that I was editing was written when we were treating unknown typecodes as shielded, rather than transparent. I agree now that us handling all known typecodes means we can remove this gate.